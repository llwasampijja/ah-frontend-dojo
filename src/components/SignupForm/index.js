// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';

// styles
import 'components/SignupForm/SignupForm.scss';

// components
import Button from 'components/Button';
import InputBox from 'components/InputBox';
// import ModalBox from 'components/ModalBox';


const SignupForm = (props) => {
  const {
    onChangeHandler, email, username, password, confirmPassword, onSubmitHandler,
  } = props;
  return (
    <div className="signupForm">
      <form>
        <InputBox
          name="email"
          inputType="email"
          handleChange={onChangeHandler}
          placeholder="email"
          inputClass="signupForm__input"
          value={email}
        >
          <span id="signupForm__input__error__email" className="signupForm__input__error">
            <small>And erro occured</small>
          </span>
        </InputBox>
        <InputBox
          name="username"
          inputType="text"
          handleChange={onChangeHandler}
          placeholder="username"
          inputClass="signupForm__input"
          value={username}
        >
          <span id="signupForm__input__error__username" className="signupForm__input__error">
            <small>And erro occured</small>
          </span>
        </InputBox>
        <InputBox
          name="password"
          inputType="password"
          handleChange={onChangeHandler}
          placeholder="password"
          inputClass="signupForm__input"
          value={password}
        >
          <span id="signupForm__input__error__password" className="signupForm__input__error">
            <small>And erro occured</small>
          </span>
        </InputBox>
        <InputBox
          name="confirmPassword"
          inputType="password"
          handleChange={onChangeHandler}
          placeholder="confirm password"
          inputClass="signupForm__input"
          value={confirmPassword}
        >
          <span id="signupForm__input__error__confirmPassword" className="signupForm__input__error">
            <small>And erro occured</small>
          </span>
        </InputBox>
        <Button
          btnClass="signupForm__button"
          btnName="Signup"
          btnEvent={onSubmitHandler}
        />
      </form>
    </div>
  );
};

// prop types validations
SignupForm.propTypes = {
  onChangeHandler: PropTypes.func,
  email: PropTypes.string,
  username: PropTypes.string,
  password: PropTypes.string,
  confirmPassword: PropTypes.string,
  onSubmitHandler: PropTypes.func,
};

SignupForm.defaultProps = {
  onChangeHandler: () => {},
  email: '',
  username: '',
  password: '',
  confirmPassword: '',
  onSubmitHandler: () => {},
};

// export default ModalBox(SignupForm);
export default SignupForm;
