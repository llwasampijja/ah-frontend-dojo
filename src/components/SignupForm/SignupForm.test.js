// react libraries
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';

// components
import SignupForm from 'components/SignupForm';

// test SignupForm component
describe('SignupForm Component', () => {
  it('matches the snapshot', () => {
    const component = shallow(<SignupForm />);
    expect(component).toMatchSnapshot();
  });
});
