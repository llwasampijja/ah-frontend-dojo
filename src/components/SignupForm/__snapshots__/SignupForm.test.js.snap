// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm Component matches the snapshot 1`] = `
<div
  className="signupForm"
>
  <form>
    <InputBox
      handleChange={[Function]}
      inputClass="signupForm__input"
      inputType="email"
      name="email"
      placeholder="email"
      required={true}
      value=""
    >
      <span
        className="signupForm__input__error"
        id="signupForm__input__error__email"
      >
        <small>
          And erro occured
        </small>
      </span>
    </InputBox>
    <InputBox
      handleChange={[Function]}
      inputClass="signupForm__input"
      inputType="text"
      name="username"
      placeholder="username"
      required={true}
      value=""
    >
      <span
        className="signupForm__input__error"
        id="signupForm__input__error__username"
      >
        <small>
          And erro occured
        </small>
      </span>
    </InputBox>
    <InputBox
      handleChange={[Function]}
      inputClass="signupForm__input"
      inputType="password"
      name="password"
      placeholder="password"
      required={true}
      value=""
    >
      <span
        className="signupForm__input__error"
        id="signupForm__input__error__password"
      >
        <small>
          And erro occured
        </small>
      </span>
    </InputBox>
    <InputBox
      handleChange={[Function]}
      inputClass="signupForm__input"
      inputType="password"
      name="confirmPassword"
      placeholder="confirm password"
      required={true}
      value=""
    >
      <span
        className="signupForm__input__error"
        id="signupForm__input__error__confirmPassword"
      >
        <small>
          And erro occured
        </small>
      </span>
    </InputBox>
    <Button
      btnClass="signupForm__button"
      btnEvent={[Function]}
      btnName="Signup"
    />
  </form>
</div>
`;

exports[`SignupUser Component matches the snapshot 1`] = `
<div>
  <SignupForm
    confirmPassword=""
    email=""
    onChangeHandler={[Function]}
    onSubmitHandler={[Function]}
    password=""
    username=""
  />
</div>
`;
